server/api/src/test/java/org/hawkular/apm/server/api/model/zipkin/SpanTest.java

```diff
@@ -71,12 +71,36 @@ public void testPartialUrl() throws MalformedURLException {
     }
 
     @Test
+    public void testUrlFromUri() throws MalformedURLException {
+        URL url = new URL("http://localhost:8080/my/path");
+        BinaryAnnotation ba = new BinaryAnnotation();
+        ba.setKey("http.uri");
+        ba.setValue(url.toString());
+
+        Span span = new Span(Arrays.asList(ba));
+
+        URL spanURL = span.url();
+        assertEquals(url, spanURL);
+    }
+
+    @Test
+    public void testUrlFromPath() throws MalformedURLException {
+        URL url = new URL("http://localhost:8080/my/path");
```

Comment:
Could you change this to be just the path, e.g. "/my/path" - to ensure it is still returned as a URL (even though missing the host and port).